aioairq/core.py

```diff
@@ -128,27 +201,88 @@ async def get_latest_data(
         return data
 
     async def get(self, subject: str) -> dict:
-        """Return the given subject from the air-Q device"""
+        """Return the given subject from the air-Q device.
+
+        This function only works on a limited set of subject specified in _supported_routes.
+        Prefer using more specialized functions."""
         if subject not in self._supported_routes:
             raise NotImplementedError(
-                f"subject must be in {self._supported_routes}, got {subject}"
+                "AirQ.get() is currently limited to a set of requests, returning "
+                f"a dict with a key 'content' (namely {self._supported_routes})."
             )
 
+        return await self._get_json_and_decode("/" + subject)
+
+    async def _get_json(self, relative_url: str) -> dict:
+        """Executes a GET request to the air-Q device with the configured timeout
+        and returns JSON data as a dictionary.
+
+        relative_url is expected to start with a slash."""
+
         async with self._session.get(
-            f"{self.anchor}/{subject}", timeout=self._timeout
+                f"{self.anchor}{relative_url}", timeout=self._timeout
         ) as response:
-            html = await response.text()
+            json_string = await response.text()
 
         try:
-            encoded_message = json.loads(html)["content"]
-        except (json.JSONDecodeError, KeyError):
+            return json.loads(json_string)
+        except json.JSONDecodeError as e:
             raise InvalidAirQResponse(
-                "AirQ.get() is currently limited to a set of requests, returning "
-                f"a dict with a key 'content' (namely {self._supported_routes}). "
-                f"AirQ.get({subject}) returned {html}"
-            )
+                "_get_json() must only be used to query endpoints returning JSON data. "
+                f"{relative_url} returned {json_string}."
+            ) from e
+
+    async def _get_json_and_decode(self, relative_url: str) -> any:
+        """Executes a GET request to the air-Q device with the configured timeout
+        decodes the response and returns JSON data.
+
+        relative_url is expected to start with a slash."""
+
+        async with self._session.get(
+                f"{self.anchor}{relative_url}", timeout=self._timeout
+        ) as response:
+            json_string = await response.text()
+
+        try:
+            json_data = json.loads(json_string)
+        except json.JSONDecodeError as e:
+            raise InvalidAirQResponse(
+                "_get_json() must only be used to query endpoints returning JSON data. "
+                f"{relative_url} returned {json_string}."
+            ) from e
```

Comment:
Or am I missing something?
```suggestion
        json_data = self._get_json(relative_url)
```

Reply:
plus `await` :wink: 

Good catch, thx. The blocks are identical.